packages/@expo/env/src/env.ts

```diff
@@ -111,9 +112,13 @@ export function createControlledEnvironment() {
 
 export function getFiles(mode: string | undefined): string[] {
   if (!mode) {
-    throw new Error(
-      'The NODE_ENV environment variable is required but was not specified. Ensure the project is bundled with Expo CLI.'
+    console.error(
+      chalk.red(
+        'The NODE_ENV environment variable is required but was not specified. Ensure the project is bundled with Expo CLI or NODE_ENV is set.'
+      )
     );
+    console.error(chalk.red('Proceeding in development mode.'));
+    mode = 'development';
```

Comment:
An alternative proposal (shared this with Evan too): if `NODE_ENV` is unset, then skip over `.env.${mode}.local` and `.env.${mode}`. Specifically, we’d allow `NODE_ENV` to be empty/unset and load just `.env.local` and `.env` in that case.